### PR TITLE
fix: always write update check timestamp if it doesn't exist

### PR DESCRIPTION
**Pull request by ysndr**
Previously, `~/.flox/cache/update-notification.json` was written after successfully checking for updates.
This means the file is never written when the CLI is up to date.
Thus periodic checks become unconditional checks executed with every run of flox,
until an update is available.

This PR distinguishes three check results:

1. skipped: the timeout for checking the file hasn't been reached yet, flox is running in a dev build, or cannot prompt to the user
2. refresh: the timeout was reached, but no update was found or no timestamp file was found.
3. update available: the timeout was reached, and an update was found

Case 2 and 3 will now write a new timestamp file to reset the cooldown for update checks.
Case 3 remains the only case that prints the update message.

### CODE REVIEW

**Comment by ysndr** [comment]
@nikitavoloboev FYI

File: cli/flox/src/commands/mod.rs [hunk 1]
```diff
@@ -445,6 +445,17 @@ enum UpdateNotificationError {
     WeMayHaveMessedUp(#[from] anyhow::Error),
 }
 
+#[derive(Debug, PartialEq)]
+enum UpdateCheckResult {
+    /// No update available, or not checked
+    Skipped,
+    /// No update available, but the notification file is missing
+    /// Used to defer file creation until the user is notified
```
Comment (mkenigs) [inline cli/flox/src/commands/mod.rs:453]: Don't we create the file even if the user is notified?
```suggestion
```

File: cli/flox/src/commands/mod.rs [hunk 2]
```diff
@@ -466,7 +477,7 @@ impl UpdateNotification {
             )
             .await
         } else {
-            Ok(None)
+            Ok(UpdateCheckResult::Skipped)
```
Comment (mkenigs) [inline cli/flox/src/commands/mod.rs:480]: nonblocking: should we skip writing the file in this case?

**Review by mkenigs — APPROVED** [review]

File: cli/flox/src/commands/mod.rs (hunk 2 above)
Comment (ysndr) [reply cli/flox/src/commands/mod.rs:480]: I mean we _are_ skipping?
do you mean we shouldn't skip?

**Comment by ysndr** [comment]
I added a `can_prompt` check to also skip if we aren't in interactive mode
@stealthybox suggested to just extend the timeout so that users with flox in their rcfiles only. For the moment i'd stick to the same approach we have elsewhere and follow up on this.

**Review by dcarley — APPROVED** [review]

File: cli/flox/src/commands/mod.rs (hunk 2 above)
Comment (mkenigs) [reply cli/flox/src/commands/mod.rs]: Nope nvm confused myself

**Pull request merged**